database/apikeys.go

```diff
@@ -45,6 +56,13 @@ func (db *DB) APIKeyCreate(ctx context.Context, user User) (*APIKeyRecord, error
 	if user.ID.IsZero() {
 		return nil, errors.New("invalid user")
 	}
+	n, err := db.staticAPIKeys.CountDocuments(ctx, bson.M{"user_id": user.ID})
+	if err != nil {
+		return nil, errors.AddContext(err, "failed to ensure user can create a new API key")
+	}
+	if n > int64(MaxNumAPIKeysPerUser) {
+		return nil, ErrMaxNumAPIKeysExceeded
```

Comment:
I would add context here that indicates `n > int64(MaxNumAPIKeysPerUser)` so the user knows that the max is configured to

Reply:
Good idea. It would make more sense to do that at the point where we communicate with the user - the programmer doesn't care about this context.